.github/workflows/build_test_deploy.yml

```diff
@@ -107,9 +107,9 @@ jobs:
     steps:
       - uses: actions/checkout@v2
       - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/*
-      - run: cat packages/next/package.json | jq '.resolutions.webpack = "^5.0.0-beta.30"' > package.json.tmp && mv package.json.tmp packages/next/package.json
-      - run: cat packages/next/package.json | jq '.resolutions.react = "^17.0.0-rc.1"' > package.json.tmp && mv package.json.tmp packages/next/package.json
-      - run: cat packages/next/package.json | jq '.resolutions."react-dom" = "^17.0.0-rc.1"' > package.json.tmp && mv package.json.tmp packages/next/package.json
+      - run: cat package.json | jq '.resolutions.webpack = "^5.0.0-beta.30"' > package.json.tmp && mv package.json.tmp package.json
+      - run: cat package.json | jq '.resolutions.react = "^17.0.0-rc.1"' > package.json.tmp && mv package.json.tmp package.json
+      - run: cat package.json | jq '.resolutions."react-dom" = "^17.0.0-rc.1"' > package.json.tmp && mv package.json.tmp package.json
```

Comment:
I noticed that the webpack 5 tests never ran, as webpack remained at 4.x on CI. Also when https://github.com/vercel/next.js/pull/17590 was merged, it looks like all tests were simply skipped:

```
Running tests: 
 test/integration/async-modules/test/index.test.js

  console.log test/jest-setup-after-env.js:5
    Configuring jest retries: 0

Test Suites: 1 skipped, 0 of 1 total
Tests:       24 skipped, 24 total
Snapshots:   0 total
Time:        4.269s
```

From https://github.com/vercel/next.js/runs/1252336307

Reply:
🤔  interesting. That means none of the webpack 5 tests are currently actually testing webpack 5 integration?

Reply:
I'm not sure, I'm really new to the repository and not familiar with the general setup. I just noticed that my added test didn't fail on CI and therefore I've adjusted this.

Reply:
Btw. the project where I originally found this bug is using the currently latest version of `webpack`: 5.3.1.

Reply:
And this doesn't happen on webpack 4?

Reply:
Top-level await is only available in webpack 5, right? I'd guess it just breaks entirely in webpack 4, but I didn't try.

Reply:
I mean, does the bundle contain `async () =>` in webpack 4, even when not using top-level-await?

Reply:
I haven't checked that yet, but what I did try is using webpack 5 and using async/await in the regular component code in `_app`. These statements were changed to ES5 as expected, but only the top-level await code resulted in native async/await code.

That made me wondering if top-level await only works with native async/await but I'd assume there should be some way for webpack to resolve the modules with promises and calling `then`.

As far as I understand the build chain, the user-land async/await code is compiled with Babel and then provided to webpack. But for top-level await code Babel doesn't attempt to transform it, since it needs to rely on the bundler to do this. The bundler chooses to turn it into native async/await which happens to break in IE11.

So I'm wondering if this is really a webpack issue, as I guess you wouldn't want to run babel again on the bundle code that webpack has produced.

Reply:
I've created [a minimal reproduction](https://github.com/amannn/webpack-top-level-await) only with webpack now and opened an issue in the webpack repository to discuss this: https://github.com/webpack/webpack/issues/11874.

Reply:
fyi, I also included these changes in https://github.com/vercel/next.js/pull/18589

Reply:
Sure, thanks!